### PR TITLE
Building of Space Traveller's App (React-Redux Group Project)

### PR DESCRIPTION
**Pull request by sameedmohsin28**
### In this project, we have
- Created descriptive readme file.
- Used React components. 
1. Rockets
2. Missions
3. My Profile
- Used React props.
- Used React Router.
- Connected React and Redux.
- Used Axios to fetch data from API.
- Handled events in the React app.
- Wrote unit tests with React Testing Library.
- Used styles in the React app.
- Used React hooks.
- Applied React best practices and language style guides in code.
- Used store, actions, and reducers in React.
- Performed code reviews for the team member.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const navigate = useNavigate();
+  const currentLocation = useLocation();
+  useEffect(() => {
+    if (currentLocation.pathname !== '/') {
+      navigate('/');
+    }
+    dispatch(rocketsApi());
+    dispatch(fetchMissions());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Comment (KDlamini) [inline src/App.js:24]: Same here, please find another to resolve the linters without disabling the rule. This is meant to help you follow the best practices.

File: src/Components/myProfile/MyProfile.jsx
Comment (KDlamini) [inline src/Components/myProfile/MyProfile.jsx:1]: [OPTIONAL]
1. The `My Profile` section can be enhanced by adding a placeholder message when the "My Missions" or "My Rockets" lists are empty (no missions joined or no rockets reserved).
1. The `My Profile` section can be enhanced by adding the "Cancel reservation" and "Leave Mission" buttons to the lists here. Clicking them should dispatch the actions you have already used in the main Rockets and Missions sections.
1. The `My Profile` section can be enhanced by adding the "Read more" button for each mission and rocket. Upon click, it should open a corresponding Wikipedia page in a new tab. NOTE - you need to get that extra Wikipedia URL from the API's payload.

File: src/Components/missions/Missions.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+              <>
+                <td><div className="notAMember">NOT A MEMBER</div></td>
+                <td><button type="submit" className="joinMission" onClick={() => dispatch(setReservationTrue(entry.mission_id))}>Join Mission</button></td>
+              </>
+            )}
+            {entry.reserved && (
+              <>
+                <td><div className="activeMember">Active Member</div></td>
+                <td><button type="submit" className="leaveMission" onClick={() => dispatch(setReservationFalse(entry.mission_id))}>Leave Mission</button></td>
+              </>
```
Comment (KDlamini) [inline src/Components/missions/Missions.jsx:34]: The button colors for `Active member` and `Not a member` on the Missions page are the other way around. If not a member the button background should be grey and when active, then it should have a bright color as per design.

Design:

image

Your design:

image

File: src/test/MissionsFetch.test.js
Comment (KDlamini) [inline src/test/MissionsFetch.test.js:1]: Kindly add more tests for Missions, there's one test for API fetching but no tests for the Missions component. 

- Kindly consider using [Jest Snapshots ](https://circleci.com/blog/snapshot-testing-with-jest/)to add more tests for all the components by adding template snapshots using Jest.

[Jest Snapshots](https://circleci.com/blog/snapshot-testing-with-jest/) is a tool used for testing React components. They help ensure that your UI changes are deterministic and that you are aware of when changes are made. Using that information, you can determine whether the changes were intended or not. A typical snapshot test case renders a UI component, takes a snapshot, then compares it to a reference snapshot file stored alongside the test. The test will fail if the two snapshots do not match: either the change is unexpected, or the reference snapshot needs to be updated to the new version of the UI component.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi team @sameedmohsin28, @Ali-0111 🙋‍♀️,  

Great work so far! 💯 
You have done a great job creating a working SpaceX Traveller's Hub with React and Redux. 🎊  
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞
 
✔️  Good PR title and description
✔️  Good API integration with Redux state manangement
✔️  Great use of `filter()` method to filter data
✔️  Impressive implementation of Rockets and Missions reservations
✔️  Navigation is smooth and seamless

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Header.jsx
[the remark this answers is not in the document]
Comment (sameedmohsin28) [reply src/components/Header.jsx:1]: @Ali-0111 
Could you please check on this issue? I tried checking it but I can not find any duplicate or repeated directories of 'component'.

[the remark this answers is not in the document]
Comment (KDlamini) [reply src/components/Header.jsx:1]: @sameedmohsin28

Please check your GitHub repo, maybe you or your partner pushed it at some point but didn't notice.

image

File: README.md [hunk 3]
```diff
@@ -1,2 +1,164 @@
...
+Allows users to
+- **reserve rockets for space travels.**
+- **join space missions.**
+- **cancel their reservations at any point of time.**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Space Traveller's Hub](https://space-travellers-sameed-ali.netlify.app/)
```
Comment (Tufahel) [inline README.md:70]: image

- [ ] ***[OPTIONAL]*** Kindly fix the live demo link from the readme file as, it is an important segment of a readme file.

**Review by Tufahel — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

Hi @sameedmohsin28 @Ali-0111  :wave: 
 
Well done on implementing all of the changes! 👍  Your project is complete now!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 👏🏻



:heavy_check_mark: Linter checks are passing.
:heavy_check_mark: Descriptive pull request.
:heavy_check_mark: Commit messages are professional.

## Optional suggestions 🤷‍♂️

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.
- You can mail me at [Mail](mailto:tufahel96@gmail.com) or ask any question at Slack, `Name: Tufahel`

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Tufahel — APPROVED** [review]
.

**Pull request merged**